app/scripts/views/components/barChart/barChart.js

```diff
@@ -306,27 +306,29 @@
       var hasGreaterOutlier = false;
       var hasSmallerOutlier = false;
       var startNumIndex = 0;
-      var endNumIndex = opts_.xDomain.length - 1;
+      var endNumIndex = opts_.xDomain.length >= 1 ? opts_.xDomain.length - 1 : 0;
```

Comment:
It won't run into line 359. Opts_.xDomain is empty means that opts_.xBarValues is also empty, so it won't run line 351 ~ 365.